README.md

```diff
@@ -217,14 +250,20 @@ We can easily queue all the jobs into the worker thread pool (see above), but th
 to continue when the work is done.  You could (but shouldn't) employ some blocking, locking, mutex +
 condition variable monstrosity, but you shouldn't.
 
+[//]: # shouldn't used reduntantly 
+
 Instead LokiMQ provides a mechanism for this by allowing you to submit a batch of jobs with a
 completion callback.  All jobs will be queued and, when the last one finishes, the finalization
 callback will be queued to continue with the task.
 
+[//]: # this sounds a lot like futures/promises, should we consider using them in the API?
```

Comment:
It is very similar, but more efficient because we do this without needing any mutexes or conditional variables (we get thread safety through the proxy-to-worker socket interaction).